### PR TITLE
ADL-37 - Add save to file / load from file features

### PR DESCRIPTION
**Pull request by nrdag**

### CODE REVIEW

File: Dashboard/Connectors/Csv/CsvConnector.cs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                    // and skip the header
+                    _reader.ReadLine();
+
+                    continue;
+                }
+
+                ParseData(line);
+
+                // Sleep for a bit
+                Thread.Sleep(1000);
```
Comment (AriadneLovelace) [inline Dashboard/Connectors/Csv/CsvConnector.cs:68]: how come sleep

File: Dashboard/Connectors/Csv/CsvConnector.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            logger.LogError(e, "Failed to open log replay file");
+        }
+
+        return null;
+    }
+
+    private void ParseData(string line)
+    {
+        var parts = line.Split(',');
+        if (parts.Length != 40)
```
Comment (AriadneLovelace) [inline Dashboard/Connectors/Csv/CsvConnector.cs:103]: how come 40?

File: DashboardTests/ConnectionTest.cs [hunk 3]
```diff
@@ -99,10 +90,7 @@ public void TestConnectionPageChangeConnectionType()
         });
 
         // Change Connection Type to IP Address
-        window.MouseDown(new Point(370, 170), MouseButton.Left);
-        window.MouseUp(new Point(370, 170), MouseButton.Left);
-        window.MouseDown(new Point(360, 220), MouseButton.Left);
-        window.MouseUp(new Point(360, 220), MouseButton.Left);
+        connectionType.SelectedItem = "IP Address";
```
Comment (AriadneLovelace) [inline DashboardTests/ConnectionTest.cs:93]: How come this one is IP Address now?

Comment (nrdag) [reply DashboardTests/ConnectionTest.cs:93]: it tried to click on ip address in the drop down and i didnt want to work out the coordinates, so i just set the value

File: Dashboard/Connectors/Csv/CsvConnector.cs (hunk 2 above)
Comment (nrdag) [reply Dashboard/Connectors/Csv/CsvConnector.cs:103]: 40 columns of data

File: Dashboard/Connectors/Csv/CsvConnector.cs (hunk 1 above)
Comment (nrdag) [reply Dashboard/Connectors/Csv/CsvConnector.cs:68]: so we don't spam the app, updates the data every second

File: DashboardTests/ConnectionTest.cs (hunk 3 above)
Comment (AriadneLovelace) [reply DashboardTests/ConnectionTest.cs:93]: fair

**Pull request merged**